src/main/java/graphql/analysis/QueryTraversal.java

```diff
@@ -100,6 +106,21 @@ private void visitImpl(QueryVisitor visitor, SelectionSet selectionSet, GraphQLC
         }
     }
 
+    protected GraphQLFieldDefinition getFieldDef(GraphQLFieldsContainer parentType, Field field) {
+        if (schema.getQueryType() == parentType) {
```

Comment:
Since we keep repeating this code, perhaps we should move it into Introspection as a utility function one day